### DISCUSSION TITLE
HTTPX -> CURL?

### DISCUSSION THREAD

**Discussion by kunaltyagi** [encode/httpx#2945] [Q&A]
For one of the API, I observed that curl and HTTPX result in different outputs. Is there any way to get the equivalent curl command from the request?

**Comment by lovelydinosaur** [comment]
Could you give us an example of how it'd work & what you'd like to see?

**Reply by kunaltyagi** [reply]
I was basically hoping for something as basic as:
```py
curl_command = request.as_curl()
# or
curl_command = httpx.utils.as_curl(request)
```
With expectation that using the request and curl_command via subprocess or print-copy-paste_to_terminal gives the same results.

I'm mostly interested in what flags would enable a 1-to-1 correspondence between https and curl
